### PR TITLE
Authed downloads

### PR DESCRIPTION
**Pull request by wasade**

### CODE REVIEW

**Comment by wasade** [comment]
@squirrelo @antgonza, possible for review?

File: amgut/lib/data_access/ag_data_access.py [hunk 1]
```diff
@@ -954,6 +954,29 @@ def handoutCheck(self, username, password):
...
+        barcode : str
+            The barcode to check access for
+
+        Returns
+        -------
+        boolean
+            True if the user can access the barcode, False otherwise
+        """
+        cursor = self.get_cursor()
+        cursor.execute("""SELECT barcode
```
Comment (antgonza) [inline amgut/lib/data_access/ag_data_access.py]: why not do something like:
cursor.execute("""SELECT EXISTS SELECT barcode FROM ag.ag_kit JOIN ag.ag_kit_barcodes USING(ag_kit_id) WHERE supplied_kit_id=%s AND barcode=%s""", [user, barcode])
return cursor.fetchone()[0]

Comment (squirrelo) [inline amgut/lib/data_access/ag_data_access.py]: :+1:

File: ag_config.txt.example [hunk 2]
```diff
@@ -8,7 +8,7 @@ TEST_ENVIRONMENT = TRUE
 
 # Path to the base directory where the data files are going to be stored
 BASE_DATA_DIR =
-
+SITEBASE =
```
Comment (squirrelo) [inline ag_config.txt.example]: This probably needs an explanation line like the other variables.

File: amgut/handlers/download.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+
+class DownloadHandler(BaseHandler):
+    @authenticated
+    def get(self, *args, **kwargs):
+        barcode = self.get_argument('barcode', None)
+        filetype = self.get_argument('filetype', None)
+
+        if barcode is None or filetype is None:
+            raise HTTPError(500, "Incorrectly formed GET request")
```
Comment (squirrelo) [inline amgut/handlers/download.py]: 400 error

File: amgut/handlers/download.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+        # Check access to file
+        has_access = ag_data.check_access(self.current_user, barcode)
+
+        if not has_access:
+            raise HTTPError(403, "Access forbidden")
+
+        if filetype not in FILETYPES:
+            raise HTTPError(500, "Unrecognized filetype")
```
Comment (squirrelo) [inline amgut/handlers/download.py]: 406 error maybe?

File: amgut/handlers/download.py [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+        filetype_path, filetype_suffix = FILETYPES[filetype]
+        fname = barcode + filetype_suffix
+        fullpath = join(filetype_path, fname)
+
+        if not exists(join(AMGUT_CONFIG.base_data_dir, fullpath)):
+            raise HTTPError(500, "File %s is not available" % fullpath)
```
Comment (squirrelo) [inline amgut/handlers/download.py]: 404 error?

**Comment by squirrelo** [comment]
Mostly just error clarification stuff.

File: ag_config.txt.example (hunk 2 above)
Comment (wasade) [inline ag_config.txt.example]: done

File: amgut/handlers/download.py (hunk 3 above)
Comment (wasade) [inline amgut/handlers/download.py]: done

File: amgut/handlers/download.py (hunk 4 above)
Comment (wasade) [inline amgut/handlers/download.py]: doesn't really seem appropriate. what about 400?

File: amgut/handlers/download.py (hunk 5 above)
Comment (wasade) [inline amgut/handlers/download.py]: on the fence, they shouldn't be able to get here if that is the case

File: amgut/lib/data_access/ag_data_access.py (hunk 1 above)
Comment (wasade) [inline amgut/lib/data_access/ag_data_access.py]: done

**Comment by antgonza** [comment]
:+1:

**Pull request merged**